### PR TITLE
chore: Cleanup ESLint warnings in CI environment

### PR DESCRIPTION
**Pull request by connorlanigan**
### Description

In a typical PR, the linting step produces a large amount of warnings with these two messages:
```
warning  Prop "className" is forbidden on Components  react/forbid-component-props
```
```
warning  Prop "id" is forbidden on Components  react/forbid-component-props
```

Example log: [cloudscape-design/components/.../job/16574616087](https://github.com/cloudscape-design/components/actions/runs/6107496245/job/16574616087?pr=1517#step:7:1)

These warnings currently make up 479 of 503 total warnings (95.2%), and thus reduce the visibility of other (potentially more important) warnings immensely. This has already lead to a "warning fatigue" in the team. Since there are still many use cases for these properties, they cannot just all be removed from the code.

Related links, issue #, if available: n/a

### How has this been tested?

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by connorlanigan — COMMENTED** [review]
Most of the file is actually unchanged, but git doesn't appear to show that correctly. I renamed `.eslintrc` to `.eslintrc.js`, and modified a few lines. I've highlighted the lines below with comments.

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+const isCI = process.env.CI;
+
+module.exports = {
```
Comment (connorlanigan) [inline .eslintrc.js:6]: These lines are new

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    '@typescript-eslint/no-unused-vars': 'error',
+    '@typescript-eslint/consistent-type-definitions': ['error', 'interface'],
+    '@typescript-eslint/explicit-module-boundary-types': 'off',
+    '@typescript-eslint/no-explicit-any': 'off',
+    '@typescript-eslint/no-non-null-assertion': 'off',
+    'react/display-name': 'off',
+    'react/no-danger': 'error',
+    'react/no-unstable-nested-components': ['error', { allowAsProps: true }],
+    'react/prop-types': 'off',
+    'react/forbid-component-props': isCI ? 'off' : ['warn', { forbid: ['className', 'id'] }],
```
Comment (connorlanigan) [inline .eslintrc.js:35]: This line is updated

Comment (just-boris) [reply .eslintrc.js:35]: Why not to open a first quick rename-PR and then another one with updates?

Comment (connorlanigan) [reply .eslintrc.js:35]: Good idea. Opened a new PR here for the renaming/format switching only: https://github.com/cloudscape-design/components/pull/1522

**Pull request closed without merging**